src/librbd/librbd.cc

```diff
@@ -1609,6 +1610,16 @@ namespace librbd {
     close();
   }
 
+  Image::Image(Image&& rhs) noexcept : ctx{std::exchange(rhs.ctx, nullptr)}
+  {
+  }
+
+  Image& Image::operator=(Image&& rhs) noexcept
+  {
+    ctx = std::exchange(rhs.ctx, nullptr);
```

Comment:
If `ctx` isn't `nullptr` (i.e. the left-hand side is opened), wouldn't this leave the underlying `ImageCtx` open forever and leak the object?

Reply:
Thanks, what an embarrassing mistake (:
Another reason not to use anything else but swap with a temporary.
Fixed that.